lib/rustler_precompiled/config.ex

```diff
@@ -83,16 +83,34 @@ defmodule RustlerPrecompiled.Config do
 
   defp validate_base_url!(nil), do: raise_for_nil_field_value(:base_url)
 
-  defp validate_base_url!(base_url) do
+  defp validate_base_url!(base_url) when is_binary(base_url) do
+    validate_base_url!({base_url, []})
+  end
+
+  defp validate_base_url!({base_url, headers}) when is_binary(base_url) and is_list(headers) do
     case :uri_string.parse(base_url) do
       %{} ->
-        base_url
+        if Enum.all?(headers, &match?({key, value} when is_list(key) and is_binary(value), &1)) do
```

Comment:
Also here, we are doing the conversion of they to charlist. Do we need to expect it as a charlist? I mean, charlists are fine, but they are not that common in Elixir APIs. So I would prefer to accept binaries.

Reply:
I assumed this was covered by tests. It turned out that the tests used direct struct initialisation and not `RustlerPrecompiled.Config.new/1`. I updated the tests to use `new/1` and sure enough, this failed a test.

Reply:
Awesome! Thank you!